Assets/Plugins/Source/Editor/EpicOnlineServicesConfigEditor.cs

```diff
@@ -225,269 +212,6 @@ private void SaveToJSONConfig(bool prettyPrint)
-            }
-            catch (OverflowException)
-            {
-            }
-            finally
-            {
-                EditorGUILayout.EndHorizontal();
-            }
-
-            EditorGUIUtility.labelWidth = originalLabelWidth;
-        }
-
-        public static void AssigningBoolField(string label, ref bool value, float labelWidth = -1, string tooltip = null)
-        {
-            float originalLabelWidth = EditorGUIUtility.labelWidth;
-            if (labelWidth >= 0)
-            {
-                EditorGUIUtility.labelWidth = labelWidth;
-            }
-
-            var newValue = EditorGUILayout.Toggle(CreateGUIContent(label, tooltip), value, GUILayout.ExpandWidth(true));
-            value = newValue;
-
-            EditorGUIUtility.labelWidth = originalLabelWidth;
-        }
-
-        public static void AssigningFloatToStringField(string label, ref string value, float labelWidth = -1, string tooltip = null)
-        {
-            float originalLabelWidth = EditorGUIUtility.labelWidth;
-            if (labelWidth >= 0)
-            {
-                EditorGUIUtility.labelWidth = labelWidth;
-            }
-
-            try
-            {
-                EditorGUILayout.BeginHorizontal();
-                var newValueAsString = EditorGUILayout.TextField(CreateGUIContent(label, tooltip), value == null ? "" : value, GUILayout.ExpandWidth(true));
-
-                if (GUILayout.Button("Clear", GUILayout.MaxWidth(50)))
-                {
-                    value = null;
-                }
-                else
-                {
-                    if (string.IsNullOrWhiteSpace(newValueAsString))
-                    {
-                        value = null;
-                        return;
-                    }
-
-                    var valueAsFloat = float.Parse(newValueAsString);
-                    value = valueAsFloat.ToString();
-                }
-            }
-            catch (FormatException)
-            {
-                value = null;
-            }
-            catch (OverflowException)
-            {
-            }
-            finally
-            {
-                EditorGUILayout.EndHorizontal();
-            }
-
-            EditorGUIUtility.labelWidth = originalLabelWidth;
-        }
-
-
-        public static void HorizontalLine(Color color)
-        {
-            var defaultHorizontalLineStyle = new GUIStyle();
-            defaultHorizontalLineStyle.normal.background = EditorGUIUtility.whiteTexture;
-            defaultHorizontalLineStyle.margin = new RectOffset(0, 0, 4, 4);
-            defaultHorizontalLineStyle.fixedHeight = 1;
-            HorizontalLine(color, defaultHorizontalLineStyle);
-        }
-
-        public static void HorizontalLine(Color color, GUIStyle guiStyle)
-        {
-            var currentColor = GUI.color;
-            GUI.color = color;
-            GUILayout.Box(GUIContent.none, guiStyle);
-            GUI.color = currentColor;
-        }
```

Comment:
All of these static methods are used to create GUI elements via UnityEditor's ImGUI based UI system. These static methods were defined in _this_ specific `EditorWindow`-derived class, but utilized by all classes that derived from `EditorWindow`, so they were moved to the new `GUIEditorHelper` static class.